internal/host/common.go

```diff
@@ -191,6 +192,22 @@ func updateRole(h *models.Host, role models.HostRole, db *gorm.DB, srcRole *stri
 		return errors.Wrapf(updateReply.Error, "failed to update host %s from cluster %s role to %s",
 			h.ID.String(), h.ClusterID.String(), role)
 	}
+
+	if IsDay2Host(h) {
+		updateReply = db.Model(h).Update("machine_config_pool_name", role)
```

Comment:
because this function updates multiple fields maybe you should use `UpdateHost` and update everything once?

Reply:
I think that everywhere we should use the same `UpdateHost`.
But then it won't take care about HostRole race-condition

Reply:
then maybe at least collect everything into a map?